veracruz-client/src/veracruz_client.rs

```diff
@@ -70,45 +71,30 @@ impl Write for InsecureConnection {
         let err = |t| std::io::Error::new(std::io::ErrorKind::Other, t);
 
         // Send all the data to the server.
-        let string_data = base64::encode(&data);
-        let combined_string = format!(
-            "{:} {:}",
+        let mut combined_data = vec![0, 0, 0, 0];
+        LittleEndian::write_u32(
```

Comment:
I don't really love this. `LittleEndian` is very arch specific. Is there another way to do this?
We've used rust `enums` with `bincode` in other places. I'm not sure if that's appropriate here, though.

Reply:
Done in other PR.